### PR TITLE
Add APM data stream rerouting docs

### PR DESCRIPTION
**Pull request by carsonip**
Closes https://github.com/elastic/apm-server/issues/12375

Feel free to make big changes! Would be helpful to know if it needs better examples / more technical depth etc

### CODE REVIEW

**Review by simitt — APPROVED** [review]
Thanks!

**Review by bmorelli25 — DISMISSED** [review]
Thanks, @carsonip!

I pushed a few small changes to better tie this addition in with our existing ingest pipeline content.

File: docs/en/observability/apm/data-streams.asciidoc [hunk 1]
```diff
@@ -85,6 +85,34 @@ Logs are stored in the following data streams:
...
+        "default"
+      ]
+    }
+  }
+]
+----
+
+For more about the reroute processor, see {ref}/reroute-processor.html[Reroute processor reference].
+
+For more information about custom ingest pipelines, see <<apm-ingest-pipelines,parse data using ingest pipelines>>.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/en/observability/apm/data-streams.asciidoc:114]: Ahhh okay, good point. I'll take a look after my next meeting and figure out how to add it back in!

[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/en/observability/apm/data-streams.asciidoc]: Okay, I moved the reroute reference link up into the content and re-added the fleet link. Thanks for pointing that out!

[the remark this answers is not in the document]
Comment (carsonip) [reply docs/en/observability/apm/data-streams.asciidoc]: lgtm, thanks

**Review by mdbirnstiehl — APPROVED** [review]
🐦

**Pull request merged**